### PR TITLE
[tckrefactor] Jakarta Persistence 3.2 - getSingleResultOrNull() to Query, TypedQuery, StoredProcedureQuery

### PR DESCRIPTION
**Pull request by rfelcman**
Another set of tests for new Jakarta Persistence 3.2. In this case it's about JPQL new features:
https://jakartaee.github.io/persistence/latest-nightly/nightly.html#jakarta-persistence-3-2

- Added getSingleResultOrNull() to Query, TypedQuery, StoredProcedureQuery

CC @lukasj @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @OndroMih @dmatej
@starksm64 @scottmarlow

### CODE REVIEW

**Comment by gurunrao** [comment]
@lukasj @scottmarlow - IMO, its good to merge the PR EOD, since we have finalize the Persistence TCK for March 29th review.

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -1,8 +0,0 @@
-<extensions>
-<extension>
-    <groupId>org.apache.maven.extensions</groupId>
-    <artifactId>maven-build-cache-extension</artifactId>
-    <version>1.0.0</version>
-</extension>
-</extensions>
-
```
Comment (lukasj) [inline .mvn/extensions.xml:8]: this should have stayed...

**Review by lukasj — COMMENTED** [review]
I also suggest to avoid tabs and use spaces

**Comment by rfelcman** [comment]
> I also suggest to avoid tabs and use spaces

OK reformatted, I used origin formatting.

**Review by gurunrao — APPROVED** [review]

**Pull request merged**